### PR TITLE
Javascript samples - Search Widget with customized GeocodeServer

### PR DESCRIPTION
**Pull request by goldenlimit**
This is a sample shows how to use Search Widget to perform search on
GeocodeServer since geocoder widget is going to be deprecated.

### CODE REVIEW

**Comment by jgravois** [comment]
itd be better if your sample leveraged a public custom geocoding service instead of one on your local machine :smile:

**Comment by goldenlimit** [comment]
I will try to find one but looks hard to "steal" someone else and that's why I create my own. Btw nice pic from babyshower :+1:

**Comment by jgravois** [comment]
here's one: http://sampleserver6.arcgisonline.com/arcgis/rest/services/Locators/Composite_HBR_Asset/GeocodeServer
i'd also recommend ~~customized~~ / **custom** geocoding service' for wording

**Comment by goldenlimit** [comment]
the reason I didn't use the sampleserver6 geocodeserver is because that server seems doesn't turn on "suggest" capability option when it published. 

I do have a case with user before that if using "Search Widget" for geocode server, user must use ArCGIS Server 10.3 or 10.3.1 and enable "Suggest" capabilities, otherwise, user may not able to see number of suggestions returned by the widget.  If user point to any GeocodeServer before ArcGIS Server 10.3, they will all get the Invalid URL error when using Search Widget.

Therefore, in order to make the application works perfectly, I need to have a GeocodeServer from 10.3 or later and with "suggestion" enabled. The one on sampleserver isn't enable suggestion I believe. 

Here is the screenshot shows that how to enable "suggest" when publishing a geocodeServer on ArcMap 10.3: 
suggestions

**Comment by jgravois** [comment]
ahh.  thanks for clarifying.

try [this](http://arclogistics.arcgisonline.com/arcgis/rest/services/King_County/GeocodeServer) one and please make an explicit note that it is a demo server and not intended for use for anything other than testing.

**Comment by goldenlimit** [comment]
@jgravois Thanks for your info! I just update the folder, please let know if you still have any question about this sample :palm_tree:

File: web-js/search-widget-with-customized-geocoder-server/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#Join stand alone table to feature layer and display data attributes in Data Grid by using the ArcGIS API for JavaScript
```
Comment (jgravois) [inline web-js/search-widget-with-customized-geocoder-server/README.md]: ?

File: web-js/search-widget-with-customized-geocoder-server/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#Join stand alone table to feature layer and display data attributes in Data Grid by using the ArcGIS API for JavaScript 
+
+This is a sample shows how to use Search Widget (esri/dijit/Search) to perform search on a customized Geocode Server since geocoder widget is going to be deprecated in the future release.
```
Comment (jgravois) [inline web-js/search-widget-with-customized-geocoder-server/README.md]: ~~customized Geocode Server~~ / custom geocoding service

File: web-js/search-widget-with-customized-geocoder-server/searchGeocodeServer.html [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      ], function (Map, FeatureLayer, Search, Locator, PictureMarkerSymbol, InfoTemplate) {
+
+         map = new Map("map", {
+            basemap: "dark-gray",
+            center: [-122.029133,47.521962], // lon, lat
+            zoom: 10
+         });
+         //ArcGIS Online feature service showing ecological footprints taken from Global FootPrint Network, 
+         //more information on this can be found http://jsapi.maps.arcgis.com/home/item.html?id=0f4c89208dce44b583d9219d843591c3
+         var layer = new FeatureLayer("http://services.arcgis.com/Ej0PsM5Aw677QF1W/arcgis/rest/services/KingCountySchoolDistricts/FeatureServer/0", {
```
Comment (jgravois) [inline web-js/search-widget-with-customized-geocoder-server/searchGeocodeServer.html]: if you're trying to demo custom geocoding services, seems unhelpful to have a feature layer too...

**Comment by goldenlimit** [comment]
@jgravois I did the change. Please let me know if I need anything to change

**Comment by jgravois** [comment]
can someone else (hopefully someone new) jump in here and run @goldenlimit 's code to confirm all works as expected and that his doc is clear?  afterward, feel free to merge.

afterward a [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) to condense all the commits is optional

**Comment by phpmaps** [comment]
I just did the test.  @goldenlimit your code is great and the sample is really helpful. The way I tested was to use the suggest REST operation.  The application suggest results and the REST results line up - telling me that the app is not suggesting things that don't exist which is a good thing.

http://arclogistics.arcgisonline.com/arcgis/rest/services/King_County/GeocodeServer/suggest?text=King&category=&location=&distance=&f=html

**Pull request merged**